run/generate.go

```diff
@@ -162,3 +163,61 @@ func doPostRun(env environ.Values, file string, postrun []string) error {
 	}
 	return nil
 }
+
+// copyStaticFiles copies files recursively from src directory to dest directory
+// while preserving the directory structure
+func copyStaticFiles(env environ.Values, src string, dest string) error {
+	if src == "" || dest == "" {
+		return nil
+	}
+	stat, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+	if !stat.IsDir() {
+		return fmt.Errorf("%s is not a directory", dest)
+	}
+	var dstat os.FileInfo
+	dstat, err = os.Stat(dest)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = os.MkdirAll(dest, stat.Mode())
+			if err != nil {
+				return err
+			}
+			dstat = stat
+		} else {
+			return err
+		}
+	}
+	if !dstat.IsDir() {
+		return fmt.Errorf("%s is not a directory", dest)
+	}
+	dirs := make(map[string]bool)
+	return filepath.Walk(src, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+		base := filepath.Dir(path)
+		rel, err := filepath.Rel(src, base)
+		if err != nil {
+			return err
+		}
+		o := filepath.Join(dest, rel)
+		if !dirs[o] {
+			err = os.MkdirAll(o, stat.Mode())
```

Comment:
I would just call mkdirall every time, it won't error out if the directory already exists, and that way you don't have to keep track